Brokerages/Samco/SamcoBrokerage.cs

```diff
@@ -0,0 +1,1072 @@
+        private readonly SamcoBrokerageAPI _samcoAPI;
+        private readonly IDataAggregator _aggregator;
+
+        /// <summary>
+        /// Locking object for the Ticks list in the data queue handler
+        /// </summary>
+        public readonly object TickLocker = new object();
+
+        private readonly CancellationTokenSource _ctsFillMonitor = new CancellationTokenSource();
+        private readonly Task _fillMonitorTask;
+        private readonly AutoResetEvent _fillMonitorResetEvent = new AutoResetEvent(false);
+        private readonly int _fillMonitorTimeout = Config.GetInt("samco.FillMonitorTimeout", 500);
+        private readonly ConcurrentDictionary<string, Order> _pendingOrders = new ConcurrentDictionary<string, Order>();
+
+        //EQUITY / COMMODITY
+        private readonly string _tradingSegment;
+        // MIS/CNC/NRML
+        private readonly string _samcoProductType;
+
+        private readonly List<string> subscribeInstrumentTokens = new List<string>();
+        private readonly List<string> unSubscribeInstrumentTokens = new List<string>();
+
+        private DateTime _lastTradeTickTime;
+
+        /// <summary>
+        /// Returns the brokerage account's base currency
+        /// </summary>
+        public override string AccountBaseCurrency => Currencies.INR;
+
+        /// <summary>
+        /// Checks if the websocket connection is connected or in the process of connecting
+        /// </summary>
+        public override bool IsConnected => WebSocket.IsOpen;
+
+        /// <summary>
+        /// Constructor for brokerage
+        /// </summary>
+        /// <param name="tradingSegment">Trading Segment</param>
+        /// <param name="productType">Product Type</param>
+        /// <param name="apiKey">api key</param>
+        /// <param name="apiSecret">api secret</param>
+        /// <param name="algorithm">the algorithm instance is required to retrieve account type</param>
+        /// <param name="yob">year of birth</param>
+        public SamcoBrokerage(string tradingSegment, string productType, string apiKey, string apiSecret, string yob, IAlgorithm algorithm, IDataAggregator aggregator)
+            : base("Samco")
+        {
+            _tradingSegment = tradingSegment;
+            _samcoProductType = productType;
+            _algorithm = algorithm;
+            _securityProvider = algorithm.Portfolio;
+            _aggregator = aggregator;
+            _samcoAPI = new SamcoBrokerageAPI();
+            _samcoAPI.Authorize(apiKey, apiSecret, yob);
+            _symbolMapper = new SamcoSymbolMapper();
+            _messageHandler = new BrokerageConcurrentMessageHandler<WebSocketMessage>(OnMessageImpl);
+
+            var subscriptionManager = new EventBasedDataQueueHandlerSubscriptionManager();
+            _algorithm.SetOptionChainProvider(new SamcoLiveOptionChainProvider(_symbolMapper));
```

Comment:
This doesn't seem right, why is this required? Can follow the pattern of other brokerages regarding this